### PR TITLE
Grappling hook

### PR DESCRIPTION
**Pull request by rplazevedo**
Added grappling hook mecahnics

### CODE REVIEW

File: Assets/Scripts/Player.cs [hunk 1]
```diff
@@ -43,4 +79,70 @@ private void Jump()
...
+    {
+        if (Input.GetMouseButtonDown(0))
+        {
+            distanceJoint.enabled = false;
+            lineRenderer.enabled = false;
+            _isGrappled = false;
+        }
+    }
+
+    void DrawLine()
```
Comment (DanielRussom) [inline Assets/Scripts/Player.cs:139]: Is this used in other files? Can it be made Private?

File: Assets/Scripts/Player.cs [hunk 2]
```diff
@@ -43,4 +79,70 @@ private void Jump()
...
+    {
+        var connectedAnchor = new Vector3(distanceJoint.connectedAnchor.x, distanceJoint.connectedAnchor.y, 0);
```
Comment (DanielRussom) [inline Assets/Scripts/Player.cs:141]: Why do we create a Vector3 here, and then seemingly create it again in ` lineRenderer.SetPosition(1, new Vector3(connectedAnchor.x, connectedAnchor.y, 0));` ? 
I might be missing something, is this doing the same thing?

File: Assets/Scripts/Player.cs (hunk 1 above)
Comment (DanielRussom) [reply Assets/Scripts/Player.cs:139]: ++ Move DrawLine above DetachGrappleOnClick (since it's called first in Grapple()) ;3

**Review by DanielRussom — CHANGES_REQUESTED** [review]
Looks good, a few questions / comments ♥

File: Assets/Scripts/Player.cs [hunk 3]
```diff
@@ -43,4 +79,70 @@ private void Jump()
...
-}
+
+    private void Grapple()
+    {   
+        // TODO: Refactor component functions
+        if (CanGrapple())
+        {
+            FireGrappleOnClick();
+        }
+        else if (_isGrappled)
```
[the remark this answers is not in the document]
Comment (rplazevedo) [reply Assets/Scripts/Player.cs:94]: I don't want to run call DrawLine if it is not grappled. Also dopn't need to check for detach grapple

File: Assets/Scripts/Player.cs (hunk 1 above)
Comment (rplazevedo) [reply Assets/Scripts/Player.cs:143]: Totatlly fixes 100% no tests needed

Comment (rplazevedo) [reply Assets/Scripts/Player.cs:143]: Source: trust me, bro

**Review by DanielRussom — APPROVED** [review]

**Pull request merged**